tracks/rust/exercises/minesweeper/mentoring.md

```diff
@@ -0,0 +1,61 @@
+### Concepts
+
+- Iterators
+- Iterator Methods
+- &str
+- char
+
+### Reasonable Solutions
+
+A reasonable solution should:
+
+- Use iterator methods instead of slice indexing whenever possible
+- Avoid unnecessary heap allocation
```

Comment:
As a learner I am still trying to acquire an intuition for what causes  needless heap allocations. In the context of this exercise I wonder if we could create an excessively large minefield to demonstrate the performance of 2 different implementations. One would use heap allocations and the other wouldn't. 
That way we could provide an avenue for the curious to see how given idioms scale.

Reply:
I'd also like to add that it would be nicer if the suggestion was a bit "SMART-er". What if a student comes with an original solution that requires quite a few allocations? Maybe suggest a certain max, like two times a N*M sized collection.